pandas/core/ops.py

```diff
@@ -1002,16 +1015,37 @@ def dispatch_to_extension_op(op, left, right):
 
     # The op calls will raise TypeError if the op is not defined
     # on the ExtensionArray
+    # TODO(jreback)
+    # we need to listify to avoid ndarray, or non-same-type extension array
+    # dispatching
+
     if is_extension_array_dtype(left):
-        res_values = op(left.values, right)
+
+        new_left = left.values
+        if (isinstance(right, np.ndarray) or
+                (is_extension_array_dtype(right) and
+                 type(left) != type(right))):
+            new_right = list(right)
+        else:
+            new_right = right
+
     else:
-        # We know that left is not ExtensionArray and is Series and right is
-        # ExtensionArray.  Want to force ExtensionArray op to get called
-        res_values = op(list(left.values), right.values)
 
+        new_left = list(left.values)
+        new_right = right
+
+    res_values = op(new_left, new_right)
     res_name = get_op_result_name(left, right)
-    return left._constructor(res_values, index=left.index,
-                             name=res_name)
+
+    # divmod returns a tuple
+    def compute_result(res_values):
```

Comment:
I think `_construct_result` and `_construct_divmod_result` exist for this purpose.

Reply:
done